### PR TITLE
docs: add details on NPM error 403 to troubleshooting doc

### PR DESCRIPTION
**Pull request by schalkneethling**
Adds clarifying information with regards to NPM error 403 as discussed in this issue: https://github.com/semantic-release/semantic-release/issues/747

### CODE REVIEW

File: docs/support/troubleshooting.md [hunk 1]
```diff
@@ -5,8 +5,9 @@
 This error happens when the commit associated with the last release cannot be found in the branch history.
 
 Multiple situation can cause this issue:
-- The package name configured in your `package.json` already exists and **semantic-release** obtains the last release of that package, which is unrelated to yours
```
Comment (pvdlg) [inline docs/support/troubleshooting.md]: Can remove this change of syntax that is unrelated to this PR?

File: docs/support/troubleshooting.md [hunk 2]
```diff
@@ -29,3 +30,18 @@ You can recover from that issue by identifying the commit in your branch history
...
+
+## You do not have permission to publish 'package-name'
+
+When running semantic-release you may encounter the error below:
+
+```
+An error occurred while running semantic-release: { Error: Command failed: npm publish ./. --registry https://registry.npmjs.org/
+npm ERR! publish Failed PUT 403
+npm ERR! code E403
+npm ERR! You do not have permission to publish "package". Are you logged in as the correct user? : package
```
Comment (pvdlg) [inline docs/support/troubleshooting.md]: Maybe use `package-name` in the error message instead of `package`? That would be clearer.

File: docs/support/troubleshooting.md [hunk 3]
```diff
@@ -29,3 +30,18 @@ You can recover from that issue by identifying the commit in your branch history
...
+```
+
+This has in fact nothing to do with your `NPM_TOKEN` or authentication method. It is actually related to this NPM issue: [https://github.com/npm/npm/issues/8561](https://github.com/npm/npm/issues/8561) and [Improve message thrown from Error code 'E403'](https://github.com/npm/registry/issues/240). The problem is that there is already a package with the same name as the one you are trying to publish. The resolution therefore is to rename your package in your `package.json`
```
Comment (pvdlg) [inline docs/support/troubleshooting.md]: I wouldn't be that categorical here. There is situation in which this error message is actually correct and caused by a permission. So I would soften the language here.

Comment (pvdlg) [inline docs/support/troubleshooting.md]: No need to reference to `https://github.com/npm/registry/issues/240` here.
Just explain the error might be due to the package name (or a very similar one) is already used.

File: docs/support/troubleshooting.md [hunk 4]
```diff
@@ -29,3 +30,18 @@ You can recover from that issue by identifying the commit in your branch history
...
+
+NOTE: If your package is called for example bob.js and there is already a package with that name. Changing it to bobjs will not resolve the problem as NPM will also reject names that are too similar.
```
Comment (pvdlg) [inline docs/support/troubleshooting.md]: No need for the note here. Also it's difficult to know what npm actually consider similar or not.
Just mention in the previous verbiage something like "the package name, or a very similar one, is already taken".

File: docs/support/troubleshooting.md (hunk 1 above)
Comment (schalkneethling) [reply docs/support/troubleshooting.md]: Oh, looks like my editor made a formatting change i.e. changed `-` to `*` So, the copy is not actually removed.

Comment (pvdlg) [reply docs/support/troubleshooting.md]: I don't understand what you mean... Please remove your modification so `-` stays `-`.

Comment (schalkneethling) [reply docs/support/troubleshooting.md]: I will revert. More specifically, I believe it might be prettier-atom that did the reformatting.

**Comment by schalkneethling** [comment]
@pvdlg Updated, thanks for the review.

File: docs/support/troubleshooting.md [hunk 5]
```diff
@@ -29,3 +30,18 @@ You can recover from that issue by identifying the commit in your branch history
...
+```
+An error occurred while running semantic-release: { Error: Command failed: npm publish ./. --registry https://registry.npmjs.org/
+npm ERR! publish Failed PUT 403
+npm ERR! code E403
+npm ERR! You do not have permission to publish "<package-name>". Are you logged in as the correct user? : <package-name>
+```
+
+This message is a little unclear, and might not have anything to with your `NPM_TOKEN` or authentication method. It might instead be related to the package name itself. If there is already a package with the same name as yours or, there is a very close match, it could trigger this error.
+
+Best way to be sure, is to search [npmjs.org](npmjs.org) using your package name. If there is a name conflict, rename your package in your `package.json`
```
Comment (gr2m) [inline docs/support/troubleshooting.md]: I think `[npmjs.org](npmjs.org)` should be `[npmjs.org](https://www.npmjs.com/)`, otherwise this is good to go, thanks :)

File: docs/support/troubleshooting.md [hunk 6]
```diff
@@ -5,6 +5,7 @@
 This error happens when the commit associated with the last release cannot be found in the branch history.
 
 Multiple situation can cause this issue:
+
```
Comment (pvdlg) [inline docs/support/troubleshooting.md]: Sorry to nitpick, but can you also remove this additional new line as it's not related to your addition?

**Comment by schalkneethling** [comment]
Thanks for the reviews everyone. This has been updated.

**Comment by pvdlg** [comment]
Thanks!
